dashboard/app/api.go

```diff
@@ -1244,15 +1249,33 @@ func needRepro(c context.Context, bug *Bug) bool {
 	return needReproForBug(c, canon)
 }
 
+var syzErrorTitleRe = regexp.MustCompile(`^SYZFAIL:|^SYZFATAL:`)
+
 func needReproForBug(c context.Context, bug *Bug) bool {
-	return bug.ReproLevel < ReproLevelC &&
-		len(bug.Commits) == 0 &&
-		bug.Title != corruptedReportTitle &&
-		bug.Title != suppressedReportTitle &&
-		config.Namespaces[bug.Namespace].NeedRepro(bug) &&
-		(bug.NumRepro < maxReproPerBug ||
-			bug.ReproLevel == ReproLevelNone &&
-				timeSince(c, bug.LastReproTime) > reproRetryPeriod)
+	bestReproLevel := ReproLevelC
+	// For some bugs there's anyway no chance to find a C repro.
+	if syzErrorTitleRe.MatchString(bug.Title) {
+		bestReproLevel = ReproLevelSyz
+	}
+	// We already have a C repro, so do new reproduction only once in a while.
+	if bug.ReproLevel >= bestReproLevel && timeSince(c, bug.LastReproTime) < reproStalePeriod {
+		return false
+	}
+	// There are already fixing commits.
+	if len(bug.Commits) > 0 {
+		return false
+	}
+	if bug.Title == corruptedReportTitle ||
+		bug.Title == suppressedReportTitle {
+		return false
+	}
+	if !config.Namespaces[bug.Namespace].NeedRepro(bug) {
+		return false
+	}
+	// Don't do too many unsuccessful reproductions.
+	return bug.NumRepro < maxReproPerBug ||
+		bug.ReproLevel == ReproLevelNone &&
+			timeSince(c, bug.LastReproTime) > reproRetryPeriod
```

Comment:
It seems that reproRetryPeriod cares about ReproLevelNone cases only.
Assume that intention was to introduce some delay between other types too.

Reply:
Actually, this was exactly the intention in the previous code and there was even a test for it (`TestNeedRepro4`).

But I agree that it doesn't seem reasonable now - C repro is a good to have thing, why not try to obtain it. Especially given the changes in this PR that handle syzkaller bugs separately (i.e. we won't demand a C repro for them).

I updated the logic and removed the old test.
@dvyukov , what do you think about it?